src/main/scala/io/getclump/Sources.scala

```diff
@@ -10,70 +11,70 @@ protected[getclump] trait Sources extends Tuples {
    * the key used to get that value.
    */
   def source[KS, V, K](fetch: KS => Future[Iterable[V]])(keyExtractor: V => K)
-                      (implicit cbf: CanBuildFrom[Nothing, K, KS]): ClumpSource[K, V] =
+                      (implicit cbf: CanBuildFrom[Nothing, K, KS], ec: ExecutionContext): ClumpSource[K, V] =
```

Comment:
Transformation of the fetch function happens at the time the ClumpSource is created, not when it is called. I think that could be awkward to move all the way back to the ClumpFetch instance.